microcloud/cmd/microcloud/main_init.go

```diff
@@ -414,6 +419,82 @@ func AddPeers(sh *service.Handler, systems map[string]InitSystem) error {
 	return nil
 }
 
+func ensureSystemsOutsideGateway(gateway string, localAddr string, systems map[string]InitSystem) error {
```

Comment:
This is a bit inefficient as `ensureSystemsOutsideGateway` parses all the system addresses twice; I'm happy to rework this if needed but I thought I'd go for readability for this first draft.